account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.di.accountmanagement.lambda;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.di.accountmanagement.entity.RemoveAccountRequest;
+import uk.gov.di.authentication.shared.entity.ErrorResponse;
+import uk.gov.di.authentication.shared.helpers.StateMachine;
+import uk.gov.di.authentication.shared.services.AuthenticationService;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.DynamoService;
+import uk.gov.di.authentication.shared.services.SessionService;
+import uk.gov.di.authentication.shared.services.ValidationService;
+
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyErrorResponse;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyResponse;
+
+public class RemoveAccountHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private final AuthenticationService authenticationService;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public RemoveAccountHandler(
+            AuthenticationService authenticationService,
+            ValidationService validationService,
+            SessionService sessionService) {
+        this.authenticationService = authenticationService;
+    }
+
+    public RemoveAccountHandler() {
+        ConfigurationService configurationService = new ConfigurationService();
+        this.authenticationService =
+                new DynamoService(
+                        configurationService.getAwsRegion(),
+                        configurationService.getEnvironment(),
+                        configurationService.getDynamoEndpointUri());
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
+            RemoveAccountRequest removeAccountRequest =
+                    objectMapper.readValue(input.getBody(), RemoveAccountRequest.class);
```

Comment:
you want to read the principal id from the request and do some basic validation to make sure the users subjects match up